specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,764 @@
+                        IID_PPV_ARGS(&webviewEnvironment)));
+                    wil::com_ptr<ICoreWebView2ContextMenuItem> newMenuItem;
+                    CHECK_FAILURE(webviewEnvironment->CreateContextMenuItem(L"Display page Uri", L"Shortcut", nullptr, COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_NORMAL, true, false, &newMenuItem));
+                    newMenuItem->add_CustomItemSelected(Callback<ICoreWebView2CustomItemSelectedEventHandler>(
+                        [this, info](
+                            ICoreWebView2* sender,
+                            IUnknown* args)
+                            {
+                                wil::unique_cotaskmem_string pageUri;
+                                CHECK_FAILURE(info->get_PageUri(&pageUri));
+                                std::wstring pageString = pageUri.get();
+                                m_appWindow->RunAsync([this, pageString]()
+                                {
+                                    MessageBox(
+                                        m_appWindow->GetMainWindow(), pageString.c_str(),
+                                        L"Display Page Uri", MB_OK);
+                                });
+                                return S_OK;
+                            })
+                            .Get(),
+                        &m_customItemSelectedToken);
+                    CHECK_FAILURE(items->AddValueAtIndex(itemsCount, newMenuItem.get()));
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_contextMenuRequestedToken);
+```
+
+## Win32 C++ Use Data to Display Custom Context Menu
+
+The developer can use the data provided in the Event arguments to display a custom context menu with entries of their choice. For this case, the developer specifies Handled to be true and requests a deferral. Deferral of this event should be completed when the user selects a context menu item (either the app developer will handle the case, or can return the selected option to the WebView) or when the end user dismisses the context menu by clicking outside of the context menu for example.
+
+ ```cpp
+    webview2->add_ContextMenuRequested(
+        Callback<ICoreWebView2ContextMenuRequestedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2ContextMenuRequestedEventArgs* args)
+            {
+                auto showMenu = [this, args]
+                {
+                    wil::com_ptr<ICoreWebView2ContextMenuItemCollection> items;
+                    CHECK_FAILURE(args->get_MenuItems(&items));
+                    CHECK_FAILURE(args->put_Handled(true));
+                    HMENU hPopupMenu = CreatePopupMenu();
+                    AddMenuItems(hPopupMenu, items);
+                    HWND hWnd;
+                    m_controller->get_ParentWindow(&hWnd);
+                    SetForegroundWindow(hWnd);
```

Comment:
Shouldn't need to call SetForegroundWindow. If you are already foreground, then it doesn't do anything. And if you're not already foreground, then you shouldn't steal foreground.